hack/generator/pkg/codegen/golden_files_test.go

```diff
@@ -153,6 +153,7 @@ func NewTestCodeGenerator(testName string, path string, t *testing.T, testConfig
 			"injectOriginalVersionFunction",
 			"injectOriginalVersionProperty",
 			"injectPropertyAssignmentFunctions",
+			pipeline.InjectHubFunctionStageId,
```

Comment:
makes me thing a pipeline should just be a type `{ string name; run(Types, (idFactory?)) Types}` and then this would be `injectHubFunction.name`

Reply:
I don't see how this would work - whether the factory methods return functions or objects, we don't have a reference to the value here, so we can't dereference it to access name. #pleaseexplainmore